doc/CGROUP_DELEGATION.md

```diff
@@ -0,0 +1,420 @@
+   to place your containers and VMs there too if you hack on managers for those.
+
+Users may define any amount of additional slices they like though, the four
+above are just the defaults.
+
+## Delegation
+
+Container managers and suchlike often want to control cgroups directly using
+the raw kernel APIs. That's entirely fine and supported, as long as proper
+*delegation* is followed. Delegation is a concept we inherited form cgroupsv2,
+but we expose it on cgroupsv1 too. Delegation means that some parts of the
+cgroup tree may be managed by different managers than others. As long as it is
+clear which manager manages which part of the tree each one can do within its
+sub-graph of the tree whatever it wants.
+
+Only sub-trees can be delegated (though whoever decides to request a sub-tree
+can delegate sub-sub-trees further to somebody else if they like
+it). Delegation takes place at a specific cgroup: in systemd there's a
+`Delegate=` property you can set for a service or scope unit. If you do, it's
+the cut-off point for systemd's cgroup management: the unit itself is managed
+by systemd, i.e. all its attributes are managed exclusively by systemd, however
+your program may create/remove sub-cgroups inside it freely, and those then
+become exclusive property of your program, systemd won't touch them — all
+attributes of *those* sub-cgroups can be manipulated freely and exclusively by
+your program.
+
+By turning on the `Delegate=` property for a scope or service you get a few
+guarantees:
+
+1. systemd won't fiddle with your sub-tree of the cgroup tree anymore. It won't
+   change attributes of any cgroups below it, nor will it create or remove any
+   cgroups thereunder, nor migrate processes across the boundaries of that
+   sub-tree as it deems useful anymore.
+
+2. If your service makes use of the `User=` functionality, then the sub-tree
+   will be `chown()`ed to the indicated user so that it can correctly create
+   cgroups below it. Note however that systemd will do that only in the unified
+   hierarchy (in unified and hybrid mode) as well as on systemd's own private
+   hierarchy (in legacy and hybrid mode). It won't pass ownership of the legacy
+   controller hierarchies. Delegation to less privileges processes is not safe
+   in cgroupsv1 (as a limitation of the kernel), hence systemd won't facilitate
+   access to it.
+
+3. Any BPF IP filter programs systemd installs will be installed with
+   `BPF_F_ALLOW_MULTI` so that your program can install additional ones.
+
+In unit files the `Delegate=` is superficially exposed as boolean. However,
+since v236 it optionally takes a list of controller names instead. If so,
+delegation is requested for listed controllers specifically. Note hat this only
+encodes a request. Depending on various parameters it might happen that your
+service actually will get fewer controllers delegated (for example, because the
+controller is not available on the current kernel or was turned off) or more.
```

Comment:
I'm wondering what happens here if a controller is enabled in the kernel but is not recognized/supported by systemd... Will it still be enabled? Is there something that can be done about it to enable it for a delegated subtree?

Reply:
i will add a paragraph exploring this explicitly